### DISCUSSION TITLE
Making PR of flutter_rust_bridge?

### DISCUSSION THREAD

**Discussion by fzyzcjy** [rustdesk/rustdesk#1378] [General]
Hi, congratulations about the code refactor into Flutter!

When wandering at the code curiously, I see this lib uses a custom version of the bridge: https://github.com/rustdesk/rustdesk/blob/ba8addbe2f7a26a99bf997f62df39810936951b9/Cargo.toml#L62 (https://github.com/SoLongAndThanksForAllThePizza/flutter_rust_bridge)

![image](https://user-images.githubusercontent.com/5236035/187009787-36d919b5-99a2-4b2f-804b-6a46ae11a6b4.png)
![image](https://user-images.githubusercontent.com/5236035/187009790-9ad9c390-ef4a-44f8-a616-91365077589f.png)

Since Rustdesk is also an open-source project, surely it is needless to talk about why it would be great if the features be PR into the upstream :)

**Comment by rustdesk** [comment]
I always encourage members pushing PR into the upstream, but final decision is up to the members themselves.

**Comment by rustdesk** [comment]
@SoLongAndThanksForAllThePizza

**Comment by SoLongAndThanksForAllThePizza** [comment]
Thank you a lot for creating flutter_rust_bridge(I love your code!), and I'm sorry for not creating a PR in time.
It was the first time for me to dig into flutter_rust_bridge's codebase when I was implementing the feature, and since I implemented it in a few hours, I wrote some really invasive code.
My plan was to reach out in flutter_rust_bridge's discussion section to ask if this feature is needed(since the documentation says using SyncReturn is not really preferred), if it's needed then I'll try to implement it in a better way and contribute to the flutter_rust_bridge(It will be an honor for me if I can contribute!)
But sadly, for personal reasons, I'm heavily occupied since June, so I don't really have time to code and contribute to RustDesk or flutter_rust_bridge since then😭
Contributing this feature to flutter_rust_bridge is still on the top of my to-do list, and it's my duty to contribute to the upstream repository, also since now I know that this feature is actually needed, I'll try my best to make time for doing this!
Sorry again for not creating a PR in time, I will try to refine my code and make a PR this weekend😊

**Reply by fzyzcjy** [reply]
You are welcome and thanks for the love of code!

> if this feature is needed(since the documentation says using SyncReturn is not really preferred)

Surely needed :) I personally also see cases when the Rust code is really fast operations (so it will not block UI thread for visible time), and we cannot use `await`.

> Sorry again for not creating a PR in time, I will try to refine my code and make a PR this weekend😊

Take your time, and looking forward to your PR!

**Reply by SoLongAndThanksForAllThePizza** [reply]
I've decided to add 'IrTypeSyncReturn' as a new IR type instead of throwing everything into IrTypeDelegate(which is what I did back in June).
If you think keeping SyncReturn simple(just throw them into IrTypeDelegate) is a better idea, please let me know😊
There's a bit more to do for adding a new IR type, I think I'll finish it next week!

**Reply by fzyzcjy** [reply]
To organize the discussions feel free to raise an issue in frb and we can discuss details there :)

That new ir type LGTM as long as we do not duplicate a lot of code, but instead reuse exising infra.

looking forward to your pr!
